client/webpack.config.js

```diff
@@ -1,112 +1,271 @@
+// polyfill Promise to make it work with node v0.10
 global.Promise = global.Promise || require('bluebird')
 
-var path = require('path');
-var webpack = require('webpack');
-var configData = require('./.config.json')
+var path = require('path')
+var webpack = require('webpack')
 var glob = require('glob')
 var _ = require('lodash')
 var CopyWebpackPlugin = require('copy-webpack-plugin')
 
-var STYLES_COMMONS_GLOB = 'app/styl/**/*.styl'
+// this config is being generated `./configure` in koding root.
+var configData = require('./.config.json')
 
 var BUILD_PATH         = path.join(__dirname, '../website/a/p/p', configData.rev)
 var CLIENT_PATH        = path.join(__dirname, '..', 'client')
 var WEBSITE_PATH       = path.join(__dirname, '..', 'website')
 var OLD_BUILDER_PATH   = path.join(CLIENT_PATH, './builder')
 var THIRD_PARTY_PATH   = path.join(CLIENT_PATH, './thirdparty')
 var ASSETS_PATH        = path.join(CLIENT_PATH, './assets')
-var COMMON_STYLES_PATH = path.join(CLIENT_PATH, STYLES_COMMONS_GLOB)
+var COMMON_STYLES_PATH = path.join(CLIENT_PATH, 'app/styl/**/*.styl')
+var PUBNUB_PATH        = path.join(THIRD_PARTY_PATH, 'pubnub.min.js')
+
+// we are gonna set NODE_ENV to either `production` or `development` to figure
+// out the compile target.
+var __DEV__ = process.env.NODE_ENV === 'development'
+var __PROD__ = process.env.NODE_ENV === 'production'
+var __TEST__ = process.env.NODE_ENV === 'test'
```

Comment:
Why? It's the common practice in all other `webpack` configs i saw out there. And what is the possible solution?